tests/components/solarlog/test_config_flow.py

```diff
@@ -117,13 +112,17 @@ async def test_form_exceptions(
     mock_solarlog_connector.test_connection.side_effect = None
     mock_solarlog_connector.test_extended_data_available.side_effect = exception2
 
-    result = await flow.async_step_user({CONF_HOST: HOST, CONF_HAS_PWD: True})
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_HOST: HOST, CONF_HAS_PWD: True}
+    )
     await hass.async_block_till_done()
 
     assert result["type"] is FlowResultType.FORM
     assert result["step_id"] == "password"
 
-    result = await flow.async_step_password({CONF_PASSWORD: "pwd"})
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_PASSWORD: "pwd"}
+    )
     await hass.async_block_till_done()
```

Comment:
You can remove all the `await hass.async_block_till_done()`s after the configure most of the time